doc(k8s-operator): document customizing enabled auto-instrumentations for nodejs with kubernetes-operator

atsu85 opened the pull request:
# The change
Current documentation states that it isn't possible to disable specific nodejs auto-instrumentation packages when using kubernetes-operator, but that isn't true after next kubernetes operator release, that allows enabling instrumentations based on `OTEL_NODE_ENABLED_INSTRUMENTATIONS` environment variable.

# [UPDATE:] Dependancy PRs released for now
This documentation change is caused by feature enabled by merged [PR](https://github.com/open-telemetry/opentelemetry-operator/pull/2745)
**that currently isn't released (hence the DRAFT status of this PR),**
and another [PR](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1953) that built the foundation for it.

svrnm commented:
@atsu85 any updates, can you take a look at my feedback? feel free to ask questions if you are unsure how to proceed

atsu85 commented on content/en/docs/kubernetes/operator/automatic.md:
```diff
@@ -282,12 +282,25 @@ must be able to receive OTLP over `grpc`. Therefore, the example uses
+This makes instrumentation easy, but can result in very slow application startup time and collecting too much or unwanted data.
+
+To enable only specific instrumentations you can use the `OTEL_NODE_ENABLED_INSTRUMENTATIONS` environment variable as documented in the
+[Node.js auto-instrumentation documentation](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/metapackages/auto-instrumentations-node#usage-auto-instrumentation).
```
link changed with [this commit](https://github.com/open-telemetry/opentelemetry.io/pull/4137/commits/436c3bacdbd3830a71c0dbf740eda070eaef6074), that is rebased on top of latest main branch and squashed into the [commit that updated k8s operator doc](https://github.com/open-telemetry/opentelemetry.io/pull/4137/commits/beab727f5e71fdf3f68ab31e6afb67676df1ac5b)

atsu85 replied:
> This should be documentation on the website and not (only) in the repository. 

Done with [this commit](https://github.com/open-telemetry/opentelemetry.io/pull/4137/commits/bb28bfaae25269912779b74b85c7ec6447104cdc). 

> Can you provide an additional PR to update this page as well?

I'd like to avoid creating separate issue and PR just for that, unless that is absolutely needed.

svrnm replied:
no separate PR is needed

atsu85 commented on content/en/docs/kubernetes/operator/automatic.md:
```diff
@@ -282,12 +282,25 @@ must be able to receive OTLP over `grpc`. Therefore, the example uses
-image you must either supply your own image that includes only the packages you
-want or use manual instrumentation.
+By default, the Node.js auto-instrumentation has all the
+[supported instrumentation libraries](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/metapackages/auto-instrumentations-node/README.md#supported-instrumentations)
```
I removed this link, to avoid duplicating information present in [Node.js auto-instrumentation documentation](/docs/languages/js/automatic/configuration/), that is linked bellow

atsu85 commented:
> can you take a look at my feedback?

@svrnm, sorry for noticing this so late and thanks for the feedback. 

Please take a look, I hope my solution is OK.

svrnm commented on content/en/docs/languages/js/automatic/configuration.md:
````diff
@@ -39,3 +39,13 @@ For example, to only enable the `env` and `host` detectors, you can set:
 OTEL_NODE_RESOURCE_DETECTORS=env,host
 ```
+
+### Excluding auto-instrumentation
+
+By default, the Node.js auto-instrumentation has all the
+[supported instrumentation libraries](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/metapackages/auto-instrumentations-node/README.md#supported-instrumentations)
+enabled.
+This makes instrumentation easy, but can result in slow application startup time and collecting too much or unwanted data.
+
+To enable only specific instrumentations you can use the `OTEL_NODE_ENABLED_INSTRUMENTATIONS` environment variable as documented in the
+[Node.js auto-instrumentation documentation](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/metapackages/auto-instrumentations-node#usage-auto-instrumentation).
````
Thanks for updating this, Also here, do not reference the documentation from the package, but bring the documentation to the website, as opentelemetry.io should be the place where documentation lives.

atsu85 replied:
> bring the documentation to the website

Regarding the `Node.js auto-instrumentation documentation` link: I overwrote this whole change based on the documentation from repo, so that link isn't needed any more.

atsu85 replied:
@svrnm,
> bring the documentation to the website

Regarding the `supported instrumentation libraries` link: I thought about inlining that as well.

My first preference is to keep it as it is, because:
* I just moved this link here from k8s-operator page
* It links to single source of truth for documentation (although in git repo, it is documentation, not code)
* [Supported libraries and frameworks](https://opentelemetry.io/docs/languages/js/automatic/#supported-libraries-and-frameworks) in the same repo also links to the same place (single source of truth)

Perhaps I should also link from here to [Supported libraries and frameworks](https://opentelemetry.io/docs/languages/js/automatic/#supported-libraries-and-frameworks) instead?

I'd prefer not to change existing correct information in this documentation project, although I wouldn't mind if someone else did it.

svrnm replied:
> It links to single source of truth for documentation (although in git repo, it is documentation, not code)

The single source of truth should be docs and not the READMEs in the code repos, we have similar motions going on for other languages & their automatic/zero-code instrumentations, but we can resolve this via a separate PR, especially since (as you stated) we have that kind of reference at different places already and this is not the purpose of this specific PR 

@open-telemetry/javascript-maintainers , @open-telemetry/docs-maintainers please take a look

atsu85 replied:
> we can resolve this via a separate PR, especially since (as you stated) we have that kind of reference at different places already and this is not the purpose of this specific PR

@svrnm , thanks! Could you also approve this PR?

swiatekm-sumo reviewed (approved):
LGTM

atsu85 commented:
FYI: amended commits to fix linting errors (only includes changes done by `npm run fix:format`)

atsu85 commented:
I saw that [Spelling check action was failing](https://github.com/open-telemetry/opentelemetry.io/actions/runs/8649751348/job/23719991432?pr=4137#step:3:18), because of  `nestjs` in the link title:
![image](https://github.com/open-telemetry/opentelemetry.io/assets/1206512/dbd82f3e-a0d2-4a08-b6d7-3c914bac41d1)
so [I replaced it with @opentelemetry/instrumentation-express](https://github.com/open-telemetry/opentelemetry.io/pull/4137/commits/ad12e3470a4e04c98e490766a72169cbf4ac643b), in hope that it shouldn't fail spell check action and rebased&squashed it on top of latest main branch

svrnm commented:
Thank you @atsu85
